File: WebApp.BFF/Core/DTOs/MessageDto.cs
using WebApp.BFF.Core.Models;

namespace WebApp.BFF.Core.DTOs
{
    public class MessageDto
    {
        public string Id { get; set; }
        public UserDto Author { get; set; }
        public string GroupId { get; set; }
        public string? Text { get; set; }
        public DateTime CreationDate { get; set; }

        internal MessageDto()
        {

        }
        public MessageDto(string id, UserDto author, string groupId, DateTime creationDate, string? text = null)
        {
            Id = id;
            Author = author;
            GroupId = groupId;
            CreationDate = creationDate;
            Text = text;
        }
    }


    public class MessageGroupByDateDto
    {
        public DateTime CreationDate { get; set; }
        public List<MessageDto> Messages { get; set; }

        internal MessageGroupByDateDto()
        {
            Messages = new List<MessageDto>();
        }

        public MessageGroupByDateDto(DateTime creationDate, List<MessageDto> messages)
        {
            CreationDate = creationDate;
            Messages = messages;
        }
    }
}

File: WebApp.SPA/ClientApp/src/components/Chatius/services/message/message.dto.ts
import { IMessageDto, IMessagesGroupedByDateDto } from "../interfaces/IMessage";
import { IUserDto } from "../interfaces/IUser";

export class MessageDto implements IMessageDto {
	id: string;
	author: IUserDto;
	groupId: string;
	text: string;
	creationDate: Date;

	constructor(
		id: string,
		author: IUserDto,
		groupId: string,
		text: string,
		creationDate: Date,
	) {
		this.id = id;
		this.author = author;
		this.groupId = groupId;
		this.text = text;
		this.creationDate = creationDate;
	}

	isSentBy(authorUsername: string): boolean {
		return this.author.userName == authorUsername;
	}

	getMessageSentDay(): string {
		const weekday = [
			"Sunday",
			"Monday",
			"Tuesday",
			"Wednesday",
			"Thursday",
			"Friday",
			"Saturday",
		];
		return weekday[new Date(this.creationDate).getDay()];
	}

	getMessageSentDate(): string {
		let date = new Date(this.creationDate);
		return date.toDateString();
	}

	getMessageSentTime(): string {
		let date = new Date(this.creationDate);
		return date.toLocaleTimeString("el-GR");
	}
}

export class MessagesGroupedByDateDto implements IMessagesGroupedByDateDto {
	creationDate: Date;
	messages: Array<IMessageDto>;

	constructor(creationDate: Date, messages: Array<IMessageDto>) {
		this.creationDate = creationDate;
		this.messages = messages;
	}
}
